Add comment with UNBOXEDONLY explanation to codegen

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40117 Add comment with UNBOXEDONLY explanation to codegen**

Differential Revision: [D22075103](https://our.internmc.facebook.com/intern/diff/D22075103/)

smessmer committed:
Add comment with UNBOXEDONLY explanation to codegen



Differential Revision: [D22075103](https://our.internmc.facebook.com/intern/diff/D22075103/)

[ghstack-poisoned]

ezyang commented on aten/src/ATen/function_wrapper.py:
```diff
@@ -93,8 +93,16 @@ def TypedDict(name, attrs, total=True):  # type: ignore
+# ops that are `use_c10_dispatcher: full` need different c++ code than ops
+# that aren't `use_c10_dispatcher: full` yet. The _UNBOXEDONLY_ variants
+# are for ops that aren't `use_c10_dispatcher: full` yet and those code templates
+# can be deleted once all ops are `use_c10_dispatcher: full`.
```
Tell people they need to update both if you modify one?

ezyang replied:
I think it may also be profitable to comment all the sites where you're doing `if` conditionals on `use_c10_dispatcher`

smessmer committed:
Update on "Add comment with UNBOXEDONLY explanation to codegen"


Differential Revision: [D22075103](https://our.internmc.facebook.com/intern/diff/D22075103/)

[ghstack-poisoned]
